tests/core/test_actions.py

```diff
@@ -344,6 +362,156 @@ async def test_remote_action_with_template_param(
     ]
 
 
+@pytest.mark.parametrize("event_class", rasa.shared.utils.common.all_subclasses(Event))
+async def test_remote_action_validate_all_event_subclasses(event_class: Type[Event]):
+    response = {"events": [{"event": event_class.type_name}], "responses": []}
+
+    from jsonschema import validate
+
+    if (
+        event_class.type_name != "wrong_utterance"
+        and event_class.type_name != "wrong_action"
+    ):
+        validate(response, RemoteAction.action_response_format_spec())
+
+
+@pytest.mark.parametrize(
+    "event",
+    (
+        EntitiesAdded(entities=[], timestamp=None),
+        DefinePrevUserUtteredFeaturization(
+            use_text_for_featurization=False, timestamp=None, metadata=None
+        ),
+        ReminderCancelled(timestamp=1621590172.3872123),
+        ReminderScheduled(
+            timestamp=None, trigger_date_time=datetime.now(), intent="greet"
+        ),
+        ActionExecutionRejected(action_name="my_action"),
+        LegacyFormValidation(validate=True, timestamp=None),
+        LoopInterrupted(timestamp=None, is_interrupted=False),
+        ActiveLoop(name="loop"),
+        LegacyForm(name="my_form"),
+        AllSlotsReset(),
+        SlotSet(key="my_slot", value={}),
```

Comment:
Can we test that with a couple of different values? E.g. lists, dicts, simple text?